harness/determined/load.py

```diff
@@ -150,11 +150,14 @@ def get_trial_controller_class(trial_class: Type[det.Trial]) -> Type[det.TrialCo
 
 def get_trial_and_controller_class(
     experiment_config: det.ExperimentConfig,
+    train_entrypoint: Optional[str],
 ) -> Tuple[Type[det.Trial], Type[det.TrialController]]:
     if experiment_config.native_enabled():
         command = experiment_config["internal"]["native"]["command"]  # type: List[str]
         trial_class = get_trial_class_from_native(command)
     else:
-        trial_class = trial_class_from_entrypoint(experiment_config["entrypoint"])
+        if not train_entrypoint:
+            raise ValueError("Entrypoint must be specified when native is disabled")
+        trial_class = trial_class_from_entrypoint(train_entrypoint)
```

Comment:
Is this helper function actually used anywhere except harness.py::main()?

It seems like the helper function has ceased to become useful.  It used to parse things from the config, but now it forces the caller to parse some things out of the config, while still requiring the config to get passed in to parse other things.  It might be cleaner to just do away with it, IMO.

Non-blocking though, feel free to disagree.

Reply:
yeah, I thought about this too. but i think the helper function does more than parse things from the config. i think it's mostly useful right now as differentiating between which type of trial class (native or not) to load.

Reply:
But actually the caller of this helper function has always already checked that case before in `launch.py`, where you have a codepath that checks for `native_enabled`.